### PR TITLE
Add the Compose language service

### PR DESCRIPTION
**Pull request by bwateratmsft**
Part of #2539.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2308,15 +2308,11 @@
                     "default": "docker",
                     "description": "%vscode-docker.config.docker.dockerPath%",
                     "scope": "machine-overridable"
-                }
-            }
-        },
-        "configurationDefaults": {
-            "[dockercompose]": {
```
Comment (bwateratmsft) [inline package.json:2315]: This may need to come back

**Comment by rcjsuen** [comment]
Congratulations and thank you for contributing to the LSP community!

**Comment by bwateratmsft** [comment]
> Congratulations and thank you for contributing to the LSP community!

Thanks @rcjsuen! We're excited. Still a long ways to go on it, but gotta start somewhere!

File: package.json [hunk 2]
```diff
@@ -2308,15 +2308,11 @@
...
-                "editor.quickSuggestions": {
-                    "other": true,
-                    "comments": false,
-                    "strings": true
+                },
+                "docker.enableDockerComposeLanguageService": {
```
Comment (karolz-ms) [inline package.json:2312]: Would it make sense to roll Compose langserv as an experiment?

I am not sure if I would keep the old completion provider etc... but maybe just default `enableDockerComposeLanguageService` to false and gradually increase the population that defaults to true.

Comment (bwateratmsft) [reply package.json:2312]: Yeah. I definitely considered it. I'm not counting it out, but it's tricky to do an experiment where the population can _choose_ to be in treatment or control; and I think it _is_ important for them to be able to choose on this.

Comment (karolz-ms) [reply package.json:2312]: Agreed, but maybe we can treat this experiment in a special way, foremost as means to turn the default to off quickly if we detect that treatment population is running into unforeseen issues. At least at the beginning.

File: package.json (hunk 1 above)
Comment (bwateratmsft) [reply package.json:2315]: I don't think so. It is controlling always-on completions which are not viable (yet).

File: package.json (hunk 2 above)
Comment (bwateratmsft) [reply package.json:2312]: I created #3208 to look into this. For now I think we should check in as-is, so we can get a build for CTI tonight.

**Review by karolz-ms — APPROVED** [review]

**Pull request merged**